instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/Serializer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import software.amazon.awssdk.core.SdkPojo;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientOption;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.protocols.core.OperationInfo;
+import software.amazon.awssdk.protocols.core.ProtocolMarshaller;
+import software.amazon.awssdk.protocols.json.AwsJsonProtocolFactory;
+import software.amazon.awssdk.utils.IoUtils;
+
+class Serializer {
+
+  private final ProtocolMarshaller<SdkHttpFullRequest> protocolMarshaller;
+
+  Serializer() {
+    protocolMarshaller =
+        AwsJsonProtocolFactory.builder()
+            .clientConfiguration(
+                SdkClientConfiguration.builder()
+                    .option(SdkClientOption.ENDPOINT, URI.create("http://empty"))
+                    .build())
+            .build()
+            .createProtocolMarshaller(
+                OperationInfo.builder()
+                    .hasPayloadMembers(true)
+                    .httpMethod(SdkHttpMethod.POST)
+                    .build());
+  }
+
+  @Nullable
+  private String serialize(SdkPojo sdkPojo) {
+    Optional<ContentStreamProvider> optional =
+        protocolMarshaller.marshall(sdkPojo).contentStreamProvider();
+    return optional
+        .map(
+            csp -> {
+              try {
+                return IoUtils.toUtf8String(csp.newStream());
```

Comment:
`csp.newStream()` is never closed

Reply:
Internally it's BAIS, which does not need to be closed (per javadoc: Closing a {@code ByteArrayInputStream} has no effect). Yet, it can be easily changed to try-with-resources to be super-super safe (for a future changes for example).